edx_django_utils/hooks/triggers.py

```diff
@@ -0,0 +1,69 @@
+"""
+Triggers for actions and filters.
+"""
+from .pipeline import run_pipeline
+from .utils import get_pipeline_configuration
+
+
+def trigger_filter(trigger_name, *args, **kwargs):
+    """
+    Function that manages the execution of filters listening on a trigger. The execution
+    follows the Pipeline pattern using the pipeline runner.
+
+    Example usage:
+        result = trigger_filter(
+            'trigger_name_used_in_hooks_config',
+            request,
+            user=user,
+        )
+        >>> result
+       {
+           'result_test_function': Object,
+           'result_test_function_2nd': Object_2nd,
+       }
+
+    Arguments:
+        trigger_name (str): determines which trigger we are listening to. It also specifies which
+        hook configuration to use when calling trigger_filter.
+
+    Returns:
+        result (dict): result of the pipeline execution, i.e the accumulated output of the filters defined in
+        the hooks configuration.
+    """
+    pipeline, is_async = get_pipeline_configuration(trigger_name)
+
+    if is_async:
+        result = run_pipeline(
+            pipeline, *args, raise_exception=True, **kwargs
+        )  # TODO: change to async call.
+    else:
+        result = run_pipeline(pipeline, *args, raise_exception=True, **kwargs)
+
+    return result
+
+
+def trigger_action(trigger_name, *args, **kwargs):
+    """
+    Function that manages the execution of actions listening on a trigger action. The execution
+    follows the Pipeline pattern using the pipeline runner.
+
+    Example usage:
+        trigger_action(
+            'trigger_name_used_in_hooks_config',
+            course_mode,
+            user=user,
+        )
+
+    Arguments:
+        trigger_name (str): determines which trigger we are listening to. It also specifies which
+        hook configuration to use when calling trigger_action.
+
+    Returns:
+        None. By definition actions don't return any value.
+    """
+    pipeline, is_async = get_pipeline_configuration(trigger_name)
+
+    if is_async:
+        run_pipeline(pipeline, *args, **kwargs)  # TODO: change to async call.
+    else:
```

Comment:
What do you think about logging something in this case?

Reply:
We could log both, with an info level? It could be useful for debugging, do you agree?

Reply:
I'm thinking about the feedback from Dave and it is still possible that we end up separating actions and filters more up to the point where this is not even possible.

Reply:
Also, this being discouraged, does not mean that it is wrong. There might be legitimate reasons for this.

Reply:
@felipemontoya can you elaborate on your first comment?